### PR TITLE
126 apollo client retry

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
When apollo client fails to connect it keeps retrying infinitely.
this adds error handling to the flow and it stops after a couple of attempts.

I tried using [RetryLink](https://www.apollographql.com/docs/react/api/link/apollo-link-retry/)
but this did not seem to stop the retry attempts

About # (link your issue here)
#126 

### Reproduce steps
1. start on local on default vite port
2. navigate to ViewCollectives page
3. see console for contineous requests

### CODE REVIEW

File: packages/app/src/hooks/apollo/useCreateMongoDbApolloClient.ts [hunk 1]
```diff
@@ -42,32 +44,52 @@ export const useCreateMongoDbApolloClient = (): ApolloClient<any> | undefined =>
...
-            fetchPolicy: 'cache-and-network',
-          },
-          query: {
-            fetchPolicy: 'cache-first',
-          },
+      // ref:https://www.apollographql.com/docs/react/data/error-handling/#advanced-error-handling-with-apollo-link
+      const errorLink = onError(({ graphQLErrors, networkError }) => {
+        if (graphQLErrors)
+          graphQLErrors.forEach(({ message, locations, path }) =>
+            console.log(`[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`)
```
Comment (sirpy) [inline packages/app/src/hooks/apollo/useCreateMongoDbApolloClient.ts:51]: .error

Comment (sirpy) [reply packages/app/src/hooks/apollo/useCreateMongoDbApolloClient.ts:51]: or warn

**Review by sirpy — CHANGES_REQUESTED** [review]
Where does it define after how many attempts it would stop?
Why is it unable to connect? do we have a mongodb issue?
This doesnt happen with subgraph?

**Comment by L03TJ3** [comment]
@sirpy The only way I found to be the retry events to configurable was by implementing RetryLink but as stated, I tried following the docs and didn't have any effect.

I just noticed this behavior because of the 'blocked-origin' when running on localhost what I mentioned in tech-sync this week

File: packages/app/src/hooks/apollo/useCreateMongoDbApolloClient.ts (hunk 1 above)
Comment (L03TJ3) [reply packages/app/src/hooks/apollo/useCreateMongoDbApolloClient.ts]: done

**Comment by L03TJ3** [comment]
@sirpy not sure about subgraph as I didn't see it happen, but added the errorLink nonetheless. it could happen

**Comment by sirpy** [comment]
@L03TJ3 
1. please provide more details of the issue. logs, how to reproduce.
When I do X I get Y but I expected Z.
2. I don't see how this solves anything, it only logs the errors, no retry no anything.

**Comment by L03TJ3** [comment]
@sirpy 
1. added reproduce steps
2. I thought it was a general issue, but after re-checking it only seems to happen to have a continuous cycle when the error is caused by a forbidden domain. If you block the URL it does stop after X attempts (was not able to figure out where this defined by default).
3. As stated. I tried adding the RetryLink for the above issue, it did not seem to have any noticeable effect. I have retried this now, and it works so apparently did something wrong the first time I tried it (added a new commit)

**Review by sirpy — APPROVED** [review]

**Pull request merged**